documents/readthedoc/docs/source/Solutions/cross_language_framework_based_gramine/Readme.md

```diff
@@ -0,0 +1,54 @@
+Cross languages framework based on Gramine
+
+Intel SGX is a software guard extended instruction set designed to
+protect user privacy. Programs running in the SGX enclave will avoid
+external snooping and tampering. However, due to the diversity of user
+needs, their programs may be in different languages, such as Python,
+Java, and even Rust, Go. While the bottom layer of Intel SGX is
+developed based on C, there is no interface for other languages, and
+Gramine officially provided by Intel only supports limited languages,
+such as Python and Java. If users want to use other languages to run
+programs in the SGX enclave, they must modify the program, which will
+bring great inconvenience to users.
+
+To reduce users' burden, we provide a Cross languages ​​framework based
+on Gramine, which provides interfaces to translate other languages to C.
+Users can use the interface provided to convert different languages
+​​into C, so users can deploy their own App in the SGX environment
+without feeling.
+
+This framework can be used in many scenarios. The following is a
+specific scenario involving the participation of multiple parties and
+the joint use of private data and programs. The participation of more
+users is also supported, but we use two users as an example to
+illustrate this process.
+
+User A and user B each have some private data locally. They want to use
+the computing resources of the remote SGX-enabled server and combine the
+data of the two parties to jointly run a specific program. The specific
+process is as follows.
+
+![image](./img/cross_language.jpeg)
+
+1.  The SGX-enabled remote server will load the program into the enclave
+    created by SGX and then generate a signature. The signature contains
+    the hash value of the program (to ensure that the program has not
+    been tampered with) and the Intel SGX certification certificate
```

Comment:
the certificate is a self-signed certificate? It needs be clear

Reply:
@YangLiang3 yes, it is a self-signed certificate. 
@hhr293 generate a signature is in compilation phase, so it is before loading program into enclave